molecule/mysql_hardening/verify_additional_tests.yml

```diff
@@ -0,0 +1,36 @@
+---
+- name: check which mysql version is used
+  community.mysql.mysql_info:
+    filter: version
+    login_unix_socket: "{{ login_unix_socket | default(omit) }}"
+  register: mysql_version
+
+- name: get all users that have no password or authentication_string on MySQL version >= 5.7.6
+  community.mysql.mysql_query:
+    query:
+      - select user,host,password from mysql.user where (length(password)=0 or password="") and (length(authentication_string)=0 or authentication_string="") and user NOT IN ('mysql.sys', 'mysqlxsys', 'mariadb.sys');
```

Comment:
in the description you sa "password or authentication_string" the query reads as "password AND authentication_string". Which one is right? I think "or" is the thing we want to check here.

Reply:
reading further, how about we only check for `authentication_string` here and let the separate `password` query on line 25 run for all versions of MySQL?

Reply:
The problem is that there are authentication methods that have a non-empty authentication string but an empty password. 
There are also authentication methods that have an empty authentication_string but not an empty password. So `or` can not be used then as this could delete legitimate users. Only if both are empty we can assume that there is no password set.
I should add some more documentation here when which possibilities can happen.